File: backend/app/schemas/share.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class ShareCreate(BaseModel):
    modality: Literal["text", "image"]
    prompt: str = Field(min_length=1)
    provider: str | None = None
    model: str | None = None
    text: str | None = None
    output_url: str | None = None
    title: str | None = None


class ShareResponse(BaseModel):
    id: str
    url: str
    preview_url: str | None = None
    modality: str
    provider: str | None
    model: str | None
    prompt: str
    text: str | None
    output_url: str | None
    title: str | None
    created_at: datetime

File: frontend/src/utils/share.js
import { api } from '../api/client'

export async function createShareLink(payload) {
  const share = await api('/api/shares', {
    method: 'POST',
    body: JSON.stringify(normalizePayload(payload))
  })
  await navigator.clipboard?.writeText(share.url)
  return share
}

function normalizePayload(payload) {
  const base = {
    modality: payload.modality,
    prompt: payload.prompt || 'Untitled generation',
    provider: payload.provider || null,
    model: payload.model || null,
    title: payload.title || null
  }

  if (payload.modality === 'image') {
    return { ...base, output_url: payload.output_url || payload.url || null }
  }

  return { ...base, text: payload.text || payload.response || payload.result || '' }
}
